tox.ini

```diff
@@ -33,20 +33,20 @@ commands =
     python setup.py sdist --dist-dir={distdir} bdist_wheel --dist-dir={distdir}
     /bin/bash -c 'python -m pip install -U --force-reinstall {distdir}/nbclient*.whl'
     /bin/bash -c 'python -m pip install -U --force-reinstall --no-deps {distdir}/nbclient*.tar.gz'
+    /bin/bash -c 'python -m pip install -e git+https://git@github.com/davidbrochart/jupyter_client.git@async_client#egg=jupyter_client'
 
 [testenv]
 # disable Python's hash randomization for tests that stringify dicts, etc
 setenv =
     PYTHONHASHSEED = 0
 passenv = *
 basepython =
-    py35: python3.5
-    py36: python3.6
-    py37: python3.7
     py38: python3.8
-    flake8: python3.6
-    manifest: python3.6
-    dist: python3.6
-    docs: python3.6
+    flake8: python3.8
+    manifest: python3.8
+    dist: python3.8
+    docs: python3.8
 deps = .[dev]
-commands = pytest -vv --maxfail=2 --cov=nbclient -W always {posargs}
+commands =
+    /bin/bash -c 'python -m pip install -e git+https://git@github.com/davidbrochart/jupyter_client.git@async_client#egg=jupyter_client'
```

Comment:
We can put that in the requirements file to test as a git path without needing to toss it in the commands. That also makes the package install cleaner for local development and indicates the branch is dependent on an unreleased code dependency.

Reply:
I tried to put it in the requirements file but never managed to make it work, that's why it ended up here (I had an error about pip not being able to parse it).

Reply:
all good, there's a way to make it happy but we can instead just focus on getting async support merged and released in jupyter_client as a pre-req for this PR to be released.